### PR TITLE
[FN] Use async code for NetworkPeerServer

### PR DESCRIPTION
**Pull request by codingupastorm**

### CODE REVIEW

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs [hunk 1]
```diff
@@ -138,28 +138,9 @@ private async Task AcceptClientsAsync()
...
-                            return null;
-                        }
-                    }).ConfigureAwait(false);
-
-                    // Raise the error.
-                    if (error != null)
-                        throw error;
+                    TcpClient tcpClient = await this.tcpListener.AcceptTcpClientAsync()
+                        .WithCancellationAsync(this.serverCancel.Token)
+                        .ConfigureAwait(false);
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs]: ConfigureAwait is already called in `WithCancellationAsync`

**Comment by MithrilMan** [comment]
@zeptin @fassadlr why it shouldn't have a configureawait?
The one in the extension doesn't matter because that's an await on the inner task.
Once you start with a ConfigureAwait you have to do that for the whole chain of awaits, otherwise you could face a deadlock

**Comment by MithrilMan** [comment]
as a reference, look at this reply from the guy who knows a lot about tasks :)
https://stackoverflow.com/a/33218022/1716620

**Comment by fassadlr** [comment]
> @zeptin @fassadlr why it shouldn't have a configureawait?
> The one in the extension doesn't matter because that's an await on the inner task.
> Once you start with a ConfigureAwait you have to do that for the whole chain of awaits, otherwise you could face a deadlock

@MithrilMan doesn't `ConfigureAwait` already get called in `WithCancellationAsync`?

**Comment by MithrilMan** [comment]
@fassadlr it doesn't matter, it has to be called on every place where you await, look linked post above

**Pull request merged**